xskillscore/core/np_deterministic.py

```diff
@@ -116,17 +130,96 @@ def _pearson_r_p_value(a, b, weights, axis, skipna):
     """
     r = _pearson_r(a, b, weights, axis, skipna)
     a = np.rollaxis(a, axis)
-    df = a.shape[0] - 2
-    t_squared = r ** 2 * (df / ((1.0 - r) * (1.0 + r)))
-    _x = df / (df + t_squared)
+    b = np.rollaxis(b, axis)
+    dof = np.apply_over_axes(np.sum, np.isnan(a * b), 0).squeeze() - 2
+    dof = np.where(dof > 1.0, dof, a.shape[0] - 2)
+    t_squared = r ** 2 * (dof / ((1.0 - r) * (1.0 + r)))
+    _x = dof / (dof + t_squared)
     _x = np.asarray(_x)
     _x = np.where(_x < 1.0, _x, 1.0)
-    _a = 0.5 * df
+    _a = 0.5 * dof
     _b = 0.5
     res = special.betainc(_a, _b, _x)
+    # reset masked values to nan
+    all_nan = np.isnan(a.mean(axis=0) * b.mean(axis=0))
+    res = np.where(all_nan, np.nan, res)
```

Comment:
Thanks for the fix
I wanted to make sure it was doing what I thought it should
Could add this to testing
https://gist.github.com/raybellwaves/3834b1005094dd748d89aaf6875ca531

Reply:
This is already included in test_masked_metric